lib/connection.js

```diff
@@ -158,7 +166,15 @@ class Connection extends EventEmitter {
     }
 
     const request = this.requestBuffer.shift();
-    const responder = request && request.pop();
```

Comment:
Add comments explaining the two paths of fetching the response callback.

Reply:
Done